server/config/config.go

```diff
@@ -474,19 +544,24 @@ func parseConfig(fileBytes []byte) (*generalConfig, error) {
 		log.Warningf("Unknown fields in config: %s", err)
 	}
 
-	if err := yaml.Unmarshal([]byte(expandedFileBytes), &sharedGeneralConfig); err != nil {
+	gc := &generalConfig{}
+	if err := yaml.Unmarshal([]byte(expandedFileBytes), gc); err != nil {
 		return nil, fmt.Errorf("Error parsing config file: %s", err)
 	}
+	sharedGeneralConfig = gc
```

Comment:
Good point.
```suggestion
	// The shared config caches the last config parsed from a file so that a call to
	// `readConfig` with an empty `configFile` can return the cached config.
	sharedGeneralConfig = gc
```